tests/TestCase/Console/ConsoleOptionParserTest.php

```diff
@@ -341,7 +355,119 @@ public function testAddOptionRequiredMissing(): void
             ]);
 
         $this->expectException(ConsoleException::class);
-        $parser->parse(['--test']);
+        $parser->parse(['--test'], $this->io);
+    }
+
+    /**
+     * test adding an option and prompting and optional options
+     */
+    public function testAddOptionWithPromptNoIo(): void
+    {
+        $parser = new ConsoleOptionParser('test', false);
+        $parser->addOption('color', [
+            'prompt' => 'What is your favorite?',
+        ]);
+        $this->expectException(ConsoleException::class);
+        $parser->parse([]);
+    }
+
+    /**
+     * test adding an option and prompting and optional options
+     */
+    public function testAddOptionWithPrompt(): void
+    {
+        $parser = new ConsoleOptionParser('test', false);
+        $parser->addOption('color', [
+            'prompt' => 'What is your favorite?',
+        ]);
+        $out = new ConsoleOutput();
+        $io = new ConsoleIo($out, new ConsoleOutput(), new ConsoleInput(['red']));
+
+        $result = $parser->parse([], $io);
+        $this->assertEquals(['color' => 'red', 'help' => false], $result[0]);
+        $messages = $out->messages();
+
+        $this->assertCount(1, $messages);
+        $expected = '<question>What is your favorite?</question>' . PHP_EOL . '> ';
+        $this->assertEquals($expected, $messages[0]);
+    }
+
+    /**
+     * test adding an option and default values
+     */
+    public function testAddOptionWithPromptAndDefault(): void
+    {
+        $parser = new ConsoleOptionParser('test', false);
+
+        $this->expectException(ConsoleException::class);
+        $this->expectExceptionMessage('You cannot set both `prompt` and `default`');
+        $parser->addOption('color', [
+            'prompt' => 'What is your favorite?',
+            'default' => 'blue',
+        ]);
+    }
+
+    /**
+     * test adding an option and prompting with cli data
+     */
+    public function testAddOptionWithPromptAndProvidedValue(): void
+    {
+        $parser = new ConsoleOptionParser('test', false);
+        $parser->addOption('color', [
+            'prompt' => 'What is your favorite?',
+        ]);
+        $out = new ConsoleOutput();
+        $io = new ConsoleIo($out, new ConsoleOutput(), new ConsoleInput(['red']));
```

Comment:
looks the `'red'` is redundant IMO , since `['--color', 'blue']` provided, and `$this->assertCount(0, $out->messages());`